src/math-translator.js

```diff
@@ -371,9 +405,387 @@ function maybeTranslateMath(math, template, lang) {
+    // because it can contain \\overline
+    // TODO: Some langs do not use \\overline, but \\dot
+    const decPart =
+       `[0-9]+|\\\\(?:overline|${katexColorMacros})[A-Z]?\\{[0-9]+\\}`;
+
+    const sep = getEscapedDecimalSeparator(lang);
+
+    // This part matches strings like `\\green{1.2}`
+    const wrappedDecimal =
+        `\\\\(?:${katexColorMacros})[A-Z]?\\{-?[0-9]+${sep}[0-9]+\\}`;
+
+    // Wrapped decimal is not needed if we capture integer and decimal part
+    // because in that case we do not care that the decimal number
+    // is wrapped as a whole
+    return capture ?
+        `(${integerPart})${sep}(${decPart})` :
+        `(?:(?:${integerPart})${sep}(?:${decPart}))|(?:${wrappedDecimal})`;
+}
+
+/**
+ * Return decimal separator for a given language.
+ * The separator is used in regex so it needs to be escaped.
+ *
+ * @param {string} lang KA locale
+ * @returns {string} Decimal separator to be passed into RegExp constructor.
+ */
+function getEscapedDecimalSeparator(lang) {
+    if (MATH_RULES_LOCALES.DECIMAL_COMMA.includes(lang)) {
+        return '\\{,\\}';
+    } else if (MATH_RULES_LOCALES.ARABIC_COMMA.includes(lang)) {
+        return '\\{،\\}';
+    } else {
+        return '\\.';
+    }
+}
+
+/**
+ * Build regex string for ranges (inside intervals)
+ * or, equivalently, cartesian coordinates (without parentheses).
+ *
+ * Matches strings such as:
+ * '-1,0'
+ * '1.2;3'
+ * '\greenD4, \blue{1{,}2}'
+ * 'x,y'
+ *
+ * We pass in the language param, because we need to match
+ * different notations for decimal numbers
+ *
+ * @param {string} lang KA locale
+ * @returns {string} string to be passed to RegExp constructor
+ */
+function getOrderedPairRegexString(lang) {
+    const katexColorMacros = `\\\\(?:${KATEX_BASE_COLORS.join('|')})[A-Z]?`;
+    // Assuming single-letter variables and numbers below 1000
+    // (without thousand separator)
+    const integer =
+      `-?[0-9]+|-?${katexColorMacros}\\{-?[0-9]+\\}|-?${katexColorMacros}[0-9]`;
+    const variable = `[a-z]|${katexColorMacros}\\{[a-z]\\}`;
+    const decimal = getDecNumberRegexString(lang,
+                /* don't include capture groups */ false);
+    const numberOrLetter = `${variable}|${decimal}|${integer}`;
+    // NOTE(danielhollas): We allow comma and semicolon for all langs
+    // as separators, even though maybe some langs use only comma.
+    // Since the US strings always have commas (I think),
+    // it's not a big deal if we are more permissive.
+    // If a translator bothered to change it to semicolon,
+    // they probably had a reason.
+    let separators = ',;';
+    if (lang === 'de') {
+        separators += '|'; // For German coordinates
+    }
+    // Support LaTeX spaces, e.g. '4~; 3' (used in e.g. French notation)
+    const space = `(?:\\\\,|~|\\s)*`;
+    const sep = `${space}[${separators}]${space}`;
+    return `\\s*(${numberOrLetter})(${sep})(${numberOrLetter})\\s*`;
+}
+
+
+/**
```

Comment:
Please provide an example of what a closed and half-closed intervals are?

Reply:
Done.